js/flatbuffers.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+
+import { flatbuffers } from "flatbuffers";
+import * as util from "./util";
+
+const globalBuilder = new flatbuffers.Builder();
+
+// Re-export some types.
+export type Offset = flatbuffers.Offset;
+export class ByteBuffer extends flatbuffers.ByteBuffer {}
+export class Builder extends flatbuffers.Builder {}
+
+// This is a wrapper around the real Builder .
+// The purpose is to reuse a single ArrayBuffer for every message.
+// We can do this because the "control" messages sent to the privileged
+// side are guarenteed to be used during the call to libdeno.send().
+export function createBuilder(): Builder {
+  let bb = globalBuilder.dataBuffer();
+  // Only create a new backing ArrayBuffer if the previous one had grown very
+  // large in capacity. This should only happen rarely.
+  if (bb.capacity() > 1024) {
```

Comment:
1kb is hardly what I would consider "very large". If we're able to reuse the buffer, why not allocate a fixed 64kb buffer?

Reply:
These messages hardly never contain much data. (The one case I can think of is `readFile`, and I'd rather that be restructured to use the second "data" parameter to libdeno.send.) I'm just trying to get rid of all the tiny little "notification" messages.

Reply:
Feels like a "large message reallocation" might be a key metric to keep an eye on/externalise?